NuRadioReco/detector/detector.py

```diff
@@ -1,995 +1,80 @@
-        
+        This method will check if the JSON file containing the detector description is set up to be used by
+        the DetectorBase or GenericDetector class and cause the correct class to be created.
+
         Parameters
         ----------
-        source : str
-            'json', 'dictionary' or 'sql'
-            default value is 'json'
-            if dictionary is specified, the dictionary passed to __init__ is used
-            if 'sql' is specified, the file 'detector_sql_auth.json' file needs to be present in this folder that
-            specifies the sql server credentials (see 'detector_sql_auth.json.sample' for an example of the syntax)
         json_filename : str
             the path to the json detector description file (if first checks a path relative to this directory, then a
             path relative to the current working directory of the user)
             default value is 'ARIANNA/arianna_detector_db.json'
+        source: str
+            'json' or 'dictionary'
+            default value is 'json'
+            If 'json' is passed, the JSON dictionary at the location specified
+            by json_filename will be used
+            If 'dictionary' is passed, the dictionary specified by the parameter
+            'dictionary' will be used
+        dictionary: dict
+            If 'dictionary' is passed to the parameter source, the dictionary
+            passed to this parameter will be used for the detector description.
         assume_inf : Bool
-            Default to True, if true forces antenna models to have infinite boundary conditions, otherwise the antenna madel will be determined by the station geometry.
+            Default to True, if true forces antenna madels to have infinite boundary conditions, otherwise the antenna
+            madel will be determined by the station geometry.
         antenna_by_depth: bool (default True)
-            if True the antenna model is determined automatically depending on the depth of the antenna. This is done by
-            appending e.g. '_InfFirn' to the antenna model name.
+            if True the antenna model is determined automatically depending on the depth of the antenna.
+            This is done by appending e.g. '_InfFirn' to the antenna model name.
             if False, the antenna model as specified in the database is used.
-        create_new: bool (default:False)
-            Can be used to force the creation of a new detector object. By default, the __init__ will anly create a new
-            object of none already exists.
         """
-        self._serialization = SerializationMiddleware()
-        self._serialization.register_serializer(DateTimeSerializer(), 'TinyDate')
-        if source == 'sql':
-            self._db = buffer_db(in_memory=True)
+        if source == 'json':
+            f = open(json_filename, 'r')
+            station_dict = json.load(f)
```

Comment:
This gets rid of the (quite convenient feature) of being able to call detector paths relative to the `detector` module path, e.g. `RNO_G/RNO_season_2021.json`. This never worked for the GenericDetector anyway, but maybe you can copy the implementation from the DetectorBase class (old Detector class)? (lines 180-188)